pallets/nfts/src/tests.rs

```diff
@@ -3876,3 +3997,1034 @@ fn clear_collection_metadata_works() {
+			));
+			assert_ok!(Nfts::check_approval(&collection_id, &None, &item_owner, &delegate));
+			assert_ok!(Nfts::check_approval(
+				&collection_id,
+				&Some(item_id),
+				&item_owner,
+				&delegate
+			));
+			// Expire approval.
+			System::set_block_number(deadline + System::block_number() + 1);
+			assert_noop!(
+				Nfts::check_approval(&collection_id, &None, &item_owner, &delegate),
+				Error::<Test>::ApprovalExpired
+			);
+			assert_noop!(
+				Nfts::check_approval(&collection_id, &Some(item_id), &item_owner, &delegate),
+				Error::<Test>::NoPermission
+			);
+			assert_ok!(Nfts::cancel_collection_approval(
+				RuntimeOrigin::signed(item_owner.clone()),
+				collection_id,
+				delegate.clone(),
+			));
+		}
+		// Approve item with deadline.
+		{
+			let deadline: BlockNumberFor<Test> = 20;
+			assert_ok!(Nfts::approve_transfer(
+				RuntimeOrigin::signed(item_owner.clone()),
+				collection_id,
+				item_id,
+				delegate.clone(),
+				Some(deadline),
+			));
+			assert_noop!(
+				Nfts::check_approval(&collection_id, &None, &item_owner, &delegate),
+				Error::<Test>::NoPermission
+			);
+			assert_ok!(Nfts::check_approval(
+				&collection_id,
+				&Some(item_id),
+				&item_owner,
+				&delegate
+			));
+			// Expire approval.
+			System::set_block_number(deadline + System::block_number() + 1);
+			assert_noop!(
+				Nfts::check_approval(&collection_id, &None, &item_owner, &delegate),
+				Error::<Test>::NoPermission
+			);
+			assert_noop!(
+				Nfts::check_approval(&collection_id, &Some(item_id), &item_owner, &delegate),
+				Error::<Test>::ApprovalExpired
+			);
+			assert_ok!(Nfts::cancel_approval(
+				RuntimeOrigin::signed(item_owner.clone()),
+				collection_id,
+				item_id,
+				delegate.clone(),
+			));
+		}
+		assert_noop!(
+			Nfts::check_approval(&collection_id, &None, &item_owner, &delegate),
+			Error::<Test>::NoPermission
+		);
+		assert_noop!(
+			Nfts::check_approval(&collection_id, &Some(item_id), &item_owner, &delegate),
+			Error::<Test>::NoPermission
+		);
+	});
+}
+
+#[test]
+fn mint_should_update_account_balance_works() {
+	new_test_ext().execute_with(|| {
+		let collection_id = 0;
+		let owner = account(1);
+
+		assert_ok!(Nfts::force_create(
+			RuntimeOrigin::root(),
+			owner.clone(),
+			default_collection_config()
+		));
+
+		assert_ok!(Nfts::mint(
+			RuntimeOrigin::signed(owner.clone()),
+			collection_id,
+			42,
+			owner.clone(),
+			None
+		));
+		assert_eq!(AccountBalance::get(collection_id, &owner), 1);
+
+		// Additive.
+		assert_ok!(Nfts::mint(
+			RuntimeOrigin::signed(owner.clone()),
+			collection_id,
+			43,
+			owner.clone(),
+			None
+		));
+		assert_eq!(AccountBalance::get(collection_id, &owner), 2);
+
+		// Mint with witness data.
+		assert_ok!(Nfts::mint(
+			RuntimeOrigin::signed(account(1)),
+			collection_id,
+			44,
+			account(2),
+			Some(MintWitness { mint_price: Some(1), ..Default::default() })
+		));
+		assert_eq!(AccountBalance::get(collection_id, account(2)), 1);
```

Comment:
I would suggest to remove as this doesn't really test anything relevant to the the AccountBalance. 

Should it not be tested when there is no deposit required?

Reply:
Removed the test.